zilliqa/src/db.rs

```diff
@@ -414,8 +414,41 @@ impl Db {
             }
         }
 
+        // Helper function used for inserting entries from memory (which backs storage trie) into state_trie db (for now it's sql-lite)
+        let db_flush =
+            |db: Arc<TrieStorage>, cache: Arc<MemoryDB>, max_batch_size: usize| -> Result<()> {
+                let mut cache_storage = cache.storage.write();
+                let mut storage_iter = cache_storage.iter();
+                debug!("Doing write to db with total items {}", cache_storage.len());
+                loop {
+                    let mut keys = Vec::with_capacity(max_batch_size);
+                    let mut values = Vec::with_capacity(max_batch_size);
+
+                    storage_iter
+                        .by_ref()
+                        .take(max_batch_size)
```

Comment:
Doesn't `TrieStorage::insert_batch` handle the batching by itself anyway? Why do we need to repeat it here?

i.e. if I pass more than ~32000 key-value pairs to `insert_batch` it splits them up by itself?